experimental/vjn/yarpl/include/reactivestreams/subscriber.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+
+#include "type_traits.h"
+
+namespace reactivestreams {
+
+template <typename T>
+class Subscription;
+
+template <typename T>
+class Subscriber {
+ public:
+  virtual ~Subscriber() = default;
+
+  virtual void on_next(const T&) {}
```

Comment:
I would provide only T&& parameter version of the method. Otherwise if T is a big type, then the application code will take the hit by copying a big type... These things should be explicit in C++ in my opinion.